src/framework/app.ts

```diff
@@ -297,15 +300,26 @@ export function createApp(appConfig?: { types?: any }): App {
           schema,
         })
 
-        const app = express()
+        const expressApp = express()
+        apolloServer.applyMiddleware({ app: expressApp })
 
-        server.applyMiddleware({ app })
+        return new Promise(resolve => {
+          const server = expressApp.listen(mergedConfig.port, () => {
+            const startMessage = mergedConfig.startMessage(mergedConfig.port)
 
-        app.listen({ port: mergedConfig.port }, () =>
-          console.log(mergedConfig.startMessage(mergedConfig.port))
-        )
+            if (startMessage.length > 0) {
+              console.log(startMessage)
+            }
 
-        sendServerReadySignalToDevModeMaster()
+            sendServerReadySignalToDevModeMaster()
+            httpServer = server
+            return resolve()
+          })
+        })
+      },
+      async stop() {
+        await apolloServer?.stop()
```

Comment:
In the following, `running` would be a mutable boolean defined higher up.

```suggestion
		if (!running) {
			return console.warn('server.stop invoked but not currently running')
		}
        await apolloServer?.stop()
		await httpServer?.close()        
       	running = false
```
Something like this, might be safer?